public/exercism_completion.bash

```diff
@@ -1,7 +1,71 @@
-__from_exercism_config() {
-  COMPREPLY=()
-  local cur=${COMP_WORDS[COMP_CWORD]}
-  COMPREPLY=($(compgen -W 'demo debug configure fetch restore submit unsubmit tracks download help' -- $cur))
+_exercism () {
+  local cur prev
+
+  COMPREPLY=()   # Array variable storing the possible completions.
+  cur=${COMP_WORDS[COMP_CWORD]}
+  prev=${COMP_WORDS[COMP_CWORD-1]}
+
+  commands="configure debug download fetch list open
+  restore skip status submit tracks unsubmit
+  upgrade help"
+  tracks="csharp cpp clojure coffeescript lisp crystal
```

Comment:
Yes that is possible. But where would that file be held? The user would also have to download it and specify it's location to the bash completion script, possibly reducing any benefit from using an external file in the first place. I am in agreement that there is a better way to store the tracks, though I am not sure how to improve it. Many bash completion scripts that I have seen simply store any options in the file.